### PR TITLE
Elm ui issue #145

### PR DESCRIPTION
**Pull request by nelsonic**
Hi @SimonLab & @iteles I've written a quick intro (tutorial) for `elm-ui`
It starts from _scratch_ and goes into a bit more depth than other `elm-ui` tutorials I've seen.

To review/read it, see: https://github.com/dwyl/learn-elm/tree/elm-ui-issue%23145/tutorials/elm-ui

### CODE REVIEW

**Comment by nelsonic** [comment]
Hi @iteles & @SimonLab, I've finished the _introduction_ to `elm-ui`. 
Rather than make _this_ PR even longer (_it's already 3.5k words..._), 
I'm going to _split_ the more advanced tutorial into a _separate_ PR. 

Please review, comment & merge this intro when you can and I will create the _next_ part shortly. 
Thanks! 🌻

File: tutorials/elm-ui/README.md [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+
+## _Why?_ 🤷 ... 😢 🌧  `|>` 😍🌈
+
+Few people _love_ writing CSS.
+Most people just want to build their App
+_without_ the headache of knowing how to position things.
+**`elm-ui`** lets you build beautiful,
+fast and responsive UIs in `elm`,
+without writing _any_ CSS!
+But `elm-ui` goes _way_ beyond _just a design system.
```
Comment (SimonLab) [inline tutorials/elm-ui/README.md]:
```suggestion
But `elm-ui` goes _way_ beyond _just_ a design system.
```

Comment (nelsonic) [reply tutorials/elm-ui/README.md]: well spotted @SimonLab thanks for fixing. 👍

File: tutorials/elm-ui/README.md [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+```
+
+These are fairly standard scoped `elm` `import` statements
+they should be familiar to anyone who has read the **Modules** section of the
+[elm guide](https://guide.elm-lang.org/webapps/modules.html).
+
+In this simple example we are using
+4 out of 9 of the available `elm-ui` modules.
+The one with the most features/functions is
+[`Elemenent`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element)
```
Comment (SimonLab) [inline tutorials/elm-ui/README.md]:
```suggestion
[`Element`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element)
```

File: tutorials/elm-ui/README.md [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+which contains everything you need for creating standard HTML elements
+including their layout and positioning.
+[`Background`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element-Background),
+[`Border`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element-Border)
+and
+[`Font`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element-Font)
+are _exactly_ what you would expect them to be,
+a collection functions related to styling those properties.
```
Comment (SimonLab) [inline tutorials/elm-ui/README.md]:
```suggestion
a collection of functions related to styling those properties.
```

File: tutorials/elm-ui/README.md [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
++ Style framework (_built on `elm-ui`_):
+https://github.com/lucamug/style-framework
++ Elm UI Explorer (Style Guide):
+https://package.elm-lang.org/packages/kalutheo/elm-ui-explorer/4.0.0/
++ Brief intro to `elm-ui`
+https://orasund.gitbook.io/elm-cookbook/frameworks-1/elm-ui-1
+
+<br /><br />
+
+## _Why NOT Just Use CSS_? 🤷
```
Comment (iteles) [inline tutorials/elm-ui/README.md]: 🙌

File: tutorials/elm-ui/README.md [hunk 5]
```diff
@@ -0,0 +1,718 @@
...
+e.g:
+http://www.csszengarden.com and https://www.cssdesignawards.com or
+CSS Super Powers: https://youtu.be/mNKz3devFAw and _drawing_ in CSS:
+https://medium.com/coding-artist/a-beginners-guide-to-pure-css-images-ef9a5d069dd2
+
+Sadly, it doesn't matter how well
+_you_ - the conscientious developer/designer - learn CSS.
+Unless you can ensure that _everyone_ in your team/company
+has the same level of skill and care in their work,
+it's _inevitable_ that over-time
```
Comment (iteles) [inline tutorials/elm-ui/README.md]:
```suggestion
it's _inevitable_ that over time
```

File: tutorials/elm-ui/README.md [hunk 6]
```diff
@@ -0,0 +1,718 @@
...
+If we attempt to invoke `rgb255 75, 192, 169`
+we will see the following error:
+
+![elm-ui-rgb-error](https://user-images.githubusercontent.com/194400/70663222-cfae5100-1c5f-11ea-8fa6-6d4437e3af14.png)
+
+We feel this is a _comparatively small_ price to pay
+given that it means colors can be checked at compile time.
+It allows us to _easily_
+implement a color picker allowing people to easily select _any_ color
+when they are [_categorising_](https://github.com/dwyl/app/issues/235)
```
Comment (iteles) [inline tutorials/elm-ui/README.md]:
```suggestion
when they are [_categorising_](https://github.com/dwyl/app/issues/235) their thoughts.
```

File: tutorials/elm-ui/README.md [hunk 7]
```diff
@@ -0,0 +1,718 @@
...
+
+If we attempt to invoke `rgb255 75, 192, 169`
+we will see the following error:
+
+![elm-ui-rgb-error](https://user-images.githubusercontent.com/194400/70663222-cfae5100-1c5f-11ea-8fa6-6d4437e3af14.png)
+
+We feel this is a _comparatively small_ price to pay
+given that it means colors can be checked at compile time.
+It allows us to _easily_
+implement a color picker allowing people to easily select _any_ color
```
Comment (iteles) [inline tutorials/elm-ui/README.md]:
```suggestion
implement a color picker allowing people to smoothly select _any_ color
```

Comment (iteles) [reply tutorials/elm-ui/README.md]: Took all my might not to amend `color` to the British spelling 😆

File: tutorials/elm-ui/README.md [hunk 8]
```diff
@@ -0,0 +1,718 @@
+# `elm-ui` a Language for _Reliable_ Layout and Interface
+
+_This_ tutorial aims to take a complete beginner -
+who has never seen any `elm-ui` -
+to a basic understanding in 10 minutes.
+
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: @iteles this is a _very_ good question.
If you feel that _other_ people will ask the same question, perhaps it worth opening as an issue: https://github.com/dwyl/learn-elm/issues
Ultimately the answer is "_No, you only need to know the basics from the Elm Guide and Syntax_"

File: tutorials/elm-ui/README.md [hunk 9]
```diff
@@ -0,0 +1,718 @@
...
+<br />
+
+## _Why?_ 🤷 ... 😢 🌧  `|>` 😍🌈
+
+Few people _love_ writing CSS.
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: @iteles that `Maybe` the case if you sit next to a front-end dev or designer all day ... 😜 
But the sheer popularity of CSS/UI Frameworks should tell you _all_ you need to know about how much the _average_ "dev" wants to write CSS ... _most_ people just want to _ship_ the _functionality_ of their app and don't want to waste time with pushing pixels around the page.

`elm-ui` doesn't _allow_ you to manipulate individual pixels so even if you wanted to waste time adjusting things to match some arbitrary photoshop design, you can't. 

Anyway, if you want to update this line to say something `else` please go for it. 👍

File: tutorials/elm-ui/README.md [hunk 10]
```diff
@@ -0,0 +1,718 @@
...
+fast and responsive UIs in `elm`,
+without writing _any_ CSS!
+But `elm-ui` goes _way_ beyond _just_ a design system.
+Since it's built using pure `elm` functions,
+it gives you ***compile-time guarantees*** that your layout/styles
+are valid. So not only is it easier/faster to build the UI,
+it makes **extending and _maintaining_** your App ***effortless***!
+
+> @dwyl we _love_ the idea of having semantic, functional and responsive UIs
+with no side-effects. <br />
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: Both spellings are valid. https://english.stackexchange.com/questions/364613/side-effects
Both are used in books. The two variant is _more popular_, but it's not more _correct_.
`Coca-Cola` is more _popular_ but it hasn't _replaced_ other soft drinks in the dictionary.

File: tutorials/elm-ui/README.md [hunk 11]
```diff
@@ -0,0 +1,718 @@
...
+who has never seen any `elm-ui` -
+to a basic understanding in 10 minutes.
+
+<br />
+
+## _Why?_ 🤷 ... 😢 🌧  `|>` 😍🌈
+
+Few people _love_ writing CSS.
+Most people just want to build their App
+_without_ the headache of knowing how to position things.
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: @iteles I will be removing this _entire_ section in a future update.

File: tutorials/elm-ui/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: We can prove this empirically by creating a survey and buying google ads to sample 10k people online. The _fact_ is that _few_ people _love_ writing CSS. And those that `do` _love_ it have invested the time to _understand_ it. Everyone `else` who has actually written code for a _living_ and just want to ship features ***HATES*** the time-suck of having to "_move the button `3px` to the right in the iPhoneX view_" to keep the designer happy. If you haven't felt the pain of spending _months_ of your life doing CSS in a company that has _zero_ process around building UI, you will never know how soul-destroying it is and how much time `elm-ui` can save.

File: tutorials/elm-ui/README.md (hunk 7 above)
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: Britain _lost_ https://en.wikipedia.org/wiki/American_Revolutionary_War so it's ***`color`***.

File: tutorials/elm-ui/README.md [hunk 12]
```diff
@@ -0,0 +1,718 @@
...
+_This_ tutorial aims to take a complete beginner -
+who has never seen any `elm-ui` -
+to a basic understanding in 10 minutes.
+
+<br />
+
+## _Why?_ 🤷 ... 😢 🌧  `|>` 😍🌈
+
+Few people _love_ writing CSS.
+Most people just want to build their App
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: This is a **`fact`**. in the same way that [Girls just want to have fun](https://youtu.be/PIb6AZdTr-A) is a _fact_. 
It's the _reason_ so many CSS (App UI) frameworks exist with all the buttons and forms created for devs to use without writing any CSS. Discussed further in https://github.com/dwyl/learn-tachyons/issues/1

File: tutorials/elm-ui/README.md [hunk 13]
```diff
@@ -0,0 +1,718 @@
...
+https://youtu.be/Ie-gqwSHQr0
+
+> "_It's all right there.
+There's no other place you have to go to modify this_."
+
+
+`elm-ui` is to `HTML` and `CSS`
+what `elm` is to `JavaScript`,
+a way of making designing web applications fun again!
+`elm-ui` offers compile-time guarantees
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: why would we _remove_ this line?

File: tutorials/elm-ui/README.md [hunk 14]
```diff
@@ -0,0 +1,718 @@
...
+that your layout and styles work as expected.
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: We could just have a bullet point list of the benefits of `elm-ui` in a future iteration and avoid all repetition completely.

File: tutorials/elm-ui/README.md [hunk 15]
```diff
@@ -0,0 +1,718 @@
...
+
+Hopefully that gives you a _taste_
+for the compiler warnings provided by `elm-ui`. <br />
+The best way to experience this further
+is to fire up your text editor and start coding!
+
+
+## _How?_ 📝
+
+In a new directory, initialise your `elm` workflow with the following command:
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply tutorials/elm-ui/README.md]: At this point installation of Elm is expected. 
Once this tutorial is _merged_ I will be re-writing the main `README.md` and linking to this `elm-ui` tutorial _after_ the person has been through the Syntax and Elm Guide 
i.e. they have Elm installed. ✅

**Comment by nelsonic** [comment]
@iteles all your typos fixes and grammar corrections have been committed. 👍 
Thanks.

**Pull request merged**